src/openassetio-core/src/utils/path/windows.hpp

```diff
@@ -0,0 +1,660 @@
+    for (std::size_t idx = 0; idx < charSet.size(); ++idx) {
+      charSet[idx] = ada::character_sets::PATH_PERCENT_ENCODE[idx];
+    }
+    // Augment the %-encode set with additional characters.
+    for (std::uint8_t charCode : {kPercentHex, kColonHex, kVerticalBarHex}) {
+      charSet[charCode / kByteSize] |= static_cast<std::uint8_t>(1 << (charCode % kByteSize));
+    }
+
+    return charSet;
+  }();
+
+  /**
+   * Check if a URL contains a percent-encoded `\` or `/`.
+   *
+   * @param url URL string to check.
+   * @return true if a percent-encoded slash was found, false otherwise.
+   */
+  [[nodiscard]] bool containsPercentEncodedSlash(const StrView& url);
+
+  /**
+   * Detect an IP6 address, and if found convert it to a valid UNC
+   * hostname.
+   *
+   * @param host hostname that is potentially an IP6 address.
+   * @return Unset optional if hostname is not an IP6 address, otherwise
+   * a valid UNC hostname pointing to the IP6 address.
+   */
+  [[nodiscard]] std::optional<Str> ip6ToValidHostname(const StrView& host);
+
+  /**
+   * Check if percent-encoding is needed for a URL path, and if so
+   * append the encoded string to a second string.
+   *
+   * @param path Path to potentially percent-encode.
+   * @param appendTo String to append result of percent-encoding. Not
+   * modified if no encoding is needed.
+   * @return true if encoding was required, false otherwise.
+   */
+  static bool maybePercentEncodeAndAppendTo(const StrView& path, Str& appendTo);
+
+  /**
+   * Set the host part on a URL object.
+   *
+   * Converts localhost to 127.0.0.1, to avoid the possibility that
+   * `file://localhost/` would be auto collapsed to `file:///` during
+   * transport, which would not then be valid when converting back to a
+   * path on Windows.
+   *
+   * @param host Hostname to set on URL object.
+   * @param url URL object o modify
+   * @return true if setting host succeeded, false if the host is
+   * invalid, according to the Ada library.
+   */
+  bool setUrlHost(const StrView& host, ada::url& url);
+};
+
+/**
+ * Parsed details of a UNC (i.e. `\\`-prefixed) path.
+ */
+struct UncDetails {
+  /// Hostname or drive letter.
+  StrView hostOrDrive;
+  /// Host share name (blank for UNC device drive paths).
+  StrView shareName;
+  /// Path in share - i.e. everything after the share name.
+  StrView sharePath;
+  /// Combined share name and path - i.e. everything after the
+  /// host/drive.
+  StrView shareNameAndPath;
+  /// Complete path excluding UNC prefix - i.e. host + share name and
+  /// path
+  StrView fullPath;
+};
+
+/**
+ * Utility for handling Windows drive and standard UNC share paths.
+ *
+ * I.e. `C:\` and `\\` but NOT `\\?\`.
+ *
+ * These paths have various normalisation rules, e.g. treating `/` as
+ * well as `\` as a path separator, and trimming some trailing chars
+ * from path segments.
+ *
+ * See https://learn.microsoft.com/en-us/dotnet/standard/io/file-path-formats
+ */
+struct NormalisedPath {
```

Comment:
Was this comment written before the split into multiple files?  Rather than a "handlers" directory there's a "path" directory, and `NormalisedPath` isn't particularly big and seems to fit well with the rest of the Windows-specific logic file.

Reply:
It wasn't. I'm mostly just concerned with file-length. I find it difficult to see to which structs the individual methods belong to when scrolling through, but I grant it's a minor taste thing, so won't press on it.

Reply:
Split up in 2c345b2c - went a bit further than just this class.  Also minor tweak then added in 02667e62